### ISSUE TITLE
False positive for E1120

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#181]
Originally reported by: **the mulhern (BitBucket: [the_mulhern](http://bitbucket.org/the_mulhern))**

---

Here's the mini example:

File junk.py:

```
#!python

class Junk(object):
    def __init__(self):
        self._junk = []

    def m(self, item):
        self._junk.append(item)

Junk = Junk()
```

File junk2.py:

```
#!python

import junk

def main():
    junk.Junk.m(2)

if __name__ == "__main__":
    main()
```

pylint reports:

E:  4, 4: No value passed for parameter 'item' in function call (no-value-for-parameter)

The problem here is the assignment,

Junk = Junk() in order to obtain a singleton Junk object which the analysis doesn't seem to recognize.

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/181

**Comment by gyermolenko** [comment]
imho this use case is not worth to be fixed

**Comment by Pierre-Sassoulas** [comment]
I agree, it's possible to implement singleton another way, You don't need to use a confusiion between Junk the class and Junk the instance. Closing in favor of #1104.
